### PR TITLE
MAT-8026 updated GetAllLibraries API to be search based on criteria

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## MADiE PR

Jira Ticket: [MAT-8026](https://jira.cms.gov/browse/MAT-8026)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included and Code coverage is verified.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibrarySearchServiceImpl.java [hunk 1]
```diff
@@ -78,10 +88,32 @@ public Page<LibraryListDTO> searchLibrariesByCriteria(
...
-      SortOperation sortOperation = sort(Sort.by(Sort.Direction.DESC, "draft", "version"));
-      GroupOperation groupOperation =
+      // Find all the libraries that matches the given Criteria and fetch unique librarySetIds
+      List<String> matchedLibrarySetIds =
+          mongoTemplate
+              .aggregate(
+                  newAggregation(
+                      lookupOperation, unwindOperation, matchOperation, group("librarySetId")),
+                  CqlLibrary.class,
+                  LibrarySetIdDTO.class)
```
Comment (sb-prateekkeerthi) [inline src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibrarySearchServiceImpl.java:101]: as it is a single field class of String type can't we directly using String.class ?

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

File: src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibrarySearchServiceImpl.java (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/cqllibraryservice/repositories/CqlLibrarySearchServiceImpl.java:101]: Mongo Aggregate returns a List of objects instead of a List of Strings, We need some sort of DTO or an object that captures id.

**Review by sb-prateekkeerthi — APPROVED** [review]
